compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/HttpFiltersDesugar.java

```diff
@@ -370,4 +395,179 @@ private BSymbol getFilterRequestFuncSymbol(BType filterType) {
+        keyLiteral.value = ANN_FIELD_PATH_PARAM_ORDER;
+        keyLiteral.type = symTable.stringType;
+
+        BLangRecordLiteral paramOrderLiteralNode = (BLangRecordLiteral) TreeBuilder.createRecordLiteralNode();
+        paramOrderLiteralNode.type = new BMapType(TypeTags.MAP, symTable.intType, symTable.mapType.tsymbol);
+
+        pathParamOrderKeyValue.key = new BLangRecordLiteral.BLangRecordKey(keyLiteral);
+        pathParamOrderKeyValue.valueExpr = paramOrderLiteralNode;
+
+        getParamMapper(resourceNode.getParameters(), value.toString()).forEach((paramName, paramIndex) -> {
+            // Create a new literal for the key.
+            BLangLiteral paramKeyLiteral = (BLangLiteral) TreeBuilder.createLiteralExpression();
+            paramKeyLiteral.value = paramName;
+            paramKeyLiteral.type = symTable.stringType;
+
+            // Create a new literal for the value.
+            BLangLiteral paramValueLiteral = new BLangLiteral();
+            paramValueLiteral.value = Long.valueOf(paramIndex);
+            paramValueLiteral.type = symTable.intType;
+
+            // Create a new key-value.
+            BLangRecordLiteral.BLangRecordKeyValue pathParamOrderEntry = new BLangRecordLiteral.BLangRecordKeyValue();
+            pathParamOrderEntry.key = new BLangRecordLiteral.BLangRecordKey(paramKeyLiteral);
+            pathParamOrderEntry.valueExpr = paramValueLiteral;
+            paramOrderLiteralNode.keyValuePairs.add(pathParamOrderEntry);
+        });
+    }
+
+    private HashMap<String, Integer> getParamMapper(List<BLangSimpleVariable> parameters, String path) {
+        HashMap<String, Integer> mapper = new HashMap<>();
+        int startIndex = 0;
+        int pointerIndex = 0;
+        while (pointerIndex < path.length()) {
+            char ch = path.charAt(pointerIndex);
+            switch (ch) {
+                case '{':
+                    startIndex = pointerIndex + 1;
+                    break;
+                case '}':
+                    String token = path.substring(startIndex, pointerIndex);
+                    IntStream.range(2, parameters.size()).filter(i -> parameters.get(i).getName().getValue()
+                            .equals(token)).findFirst().ifPresent(i -> mapper.put(token, i));
+                    break;
+            }
+            pointerIndex++;
+        }
+        return mapper;
+    }
+
+    /**
+     * Return the symbol associated with the given name in the give package regardless of its package visibility.
+     *
+     * @param symResolver symbol resolver
+     * @param pos         symbol position
+     * @param env         current symbol environment
+     * @param pkgAlias    package alias
+     * @param name        symbol name
+     * @param expSymTag   expected symbol type/tag
+     * @return resolved symbol
+     */
+    private BSymbol lookupSymbolInPackage(SymbolResolver symResolver, DiagnosticPos pos, SymbolEnv env,
```

Comment:
Why can't we use the method in symResolver instead of the new one?

Reply:
This is a custom lookupSymbolInPackage method which does not consider the visibility of particular symbol. `ParamOrderConfig` which get desugared hers is an internal detail which should not be publicly available.